### PR TITLE
[BD-6] Compliance with OEP-18

### PR DESCRIPTION
**Pull request by ericfab179**
Update setup.py install_requirements definition.
Update travis file using new requirements.
Update tox file using new requirements.
Create requirements folder.
create pip_tools, base, test, travis and tox pip-tools requirements files.
Create Makefile and add upgrade command.
Include requirements files generated using upgrade command.
Update openedx.yml OEPs list.
Add venv to gitignore.

**Reviewers**
- [ ] @awais786 
- [x] @morenol
- [ ] @jmbowman

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ericfab179! I've created [OSPR-4444](https://openedx.atlassian.net/browse/OSPR-4444) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: requirements/travis.in [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Requirements for running tests in Travis
+-c constraints.txt
+
+-r test.txt
```
Comment (morenol) [inline requirements/travis.in:4]: this is not needed here

File: requirements/constraints.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# link to other information that will help people in the future to remove the
+# pin when possible.  Writing an issue against the offending project and
+# linking to it here is good.
+
+# TODO: Many pinned dependencies should be unpinned and/or moved to this constraints file.
+
+# pip 5.0.0 only works with pip 20+ and all our VMs and containers are still installing 19.3.1 by default
+pip-tools<5.0.0
+
+# Fix tox
```
Comment (morenol) [inline requirements/constraints.txt:16]: Could add a more descriptive reason to add this constraint? if you are not sure about that you should mention that it was a constraint that was added before the implementation of oep-18

Comment (ericfab179) [reply requirements/constraints.txt:16]: @morenol I am not even sure if this constraint is actually required. Look this commit https://github.com/edx/openedx-calc/commit/9b662aca5f41c5aeff3b1c81f019f2d33ffa72ee
After that some constraints were removed https://github.com/edx/openedx-calc/commit/ea6f5e4d749277279078a7a8ba490a91dbddd200. May I just remove this constraint ?

Comment (morenol) [reply requirements/constraints.txt:16]: Let's try that

**Comment by morenol** [comment]
@ericfab179 Please squash this. After that I think that it is ready.

**Comment by ericfab179** [comment]
@awais786 We think that this PR is ready to be merged.

**Comment by edx-webhook** [comment]
@ericfab179 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/openedx-calc&entry.867055334=https://github.com/edx/openedx-calc/pull/20&entry.1484655318=https://github.com/ericfab179&entry.752974735=2020-05-01+02:25&entry.1917517419=2020-05-11+16:08&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**